python/ray/tune/tune.py

```diff
@@ -341,8 +342,26 @@ def run(
     if num_samples == -1:
         num_samples = sys.maxsize
 
+    result_buffer_length = None
+    if isinstance(scheduler, HyperBandScheduler):
```

Comment:
Should we rely on an isinstance check, or should this behavior be controlled by some attribute in the class? The latter approach may be more flexible (if we eg. want to use this behavior for some future non-HB based schedulers).

Reply:
Hm good point, a property makes more sense. I'll update this soon